tls/s2n_client_extensions.c

```diff
@@ -104,22 +104,26 @@ int s2n_client_extensions_send(struct s2n_connection *conn, struct s2n_stuffer *
     const struct s2n_cipher_preferences *cipher_preferences;
     GUARD(s2n_connection_get_cipher_preferences(conn, &cipher_preferences));
 
-    if (s2n_is_ecc_enabled(cipher_preferences)) {
+    const uint8_t ecc_extension_required = s2n_ecc_extension_required(cipher_preferences);
+    if (ecc_extension_required) {
```

Comment:
Just if (s2n_ecc_extension_required(cipher_preferences)) ?

Reply:
I reuse these down below when actually writing the value. I figured the trade off of 2 uint8_t on the stack was worth it vs looping though the cipher preference again.

Reply:
> I reuse these down below when actually writing the value. I figured the trade off of 2 uint8_t on the stack was worth it vs looping though the cipher preference again.

Good catch.